### ISSUE TITLE
[Other - Accessibility] Screen reader users may entirely miss legal text. (00.00.3)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#56080]
#### VFS team name
Benefits Team 1

#### VFS product name
Supplemental Claims Form 20-0995

#### VFS feature name
Supplemental Claims Form 20-0995

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
No VA.gov Experience Standard for the issue found.

### Issue Details
Step 3 of 4: New and relevant evidence - We need your authorization to request medical records, I'm concerned that a screen reader user tabbing through the form might miss some important legal context entirely. There are a couple of points in the workflow where a page mostly consists of text that could be missed, but I get more nervous when it's legally-significant text.

If I'm a screen reader user landing on this page, the first thing I hear is "Heading level 3, We need your authorization to request your medical records." If I press [Tab] to move to the next focusable element, the next thing I hear is "Checkbox, I acknowledge and authorize this release of information." That sequence could be mistaken for a complete transaction -- VA needs X, user says okay here's X. But that sequence hides important information:
- The scope of records VA may request
- Conditions under which it doesn't make sense to proceed ("You should not complete this form unless...")
- Potential costs the user may incur ("In accordance with... VA will not pay any fees charged...")
- Changes to health information privacy protections
- How to revoke authorization

That's all important stuff!

This is a tricky issue because everything is coded accessibly and none of this is a WCAG violation. But a screen reader user behaving the way we would expect a screen reader user to behave when working through a form like this may end up being completely unaware of all of these details.

### VA.gov Experience Standard
[Category Number 00, Issue Number 00](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Platform Recommendation
This is similar to #55699 logged for another product in a slightly different context, and @humancompanion-usds is exploring a few options for how to proceed that could be codified as patterns in the design system. 

Until a better pattern for legal text is established, I would recommend the same backup plan that we decided on for that product: Associate the whole giant legal text with the "I acknowledge and authorize" checkbox using `aria-describedby`. It's not a great solution and has some known flaws, but it does ensure that a screen reader user tabbing straight to the checkbox hears some portion of the legal text and can then decide whether to explore the page further to get those details.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by joshkimux** [comment]
@rocio-desantiago wonderful work! It looks like we may be blocked on this given some design system quirks when it comes to the shadow dom.

@jamigibbs has a really cool solution [we could use here from text inputs](https://github.com/department-of-veterans-affairs/va.gov-team/issues/24502), but I don't believe it exists yet for checkboxes. Will keep updating this ticket as we learn more information on the next best step forward.

**Comment by shiragoodman** [comment]
@joshkimux @saderagsdale was this ticket resolved or is it blocked per Josh's latest comment. If it's blocked, I would request that we reopen per our "VFS Guidance" section above. Thank you!

**Comment by joshkimux** [comment]
Thanks @shiragoodman ! And this is on me @saderagsdale , I'm keeping an eye on all of our blocked tickets on our accessibility dashboard (there's 4 currently 😄 ). This won't show up as an open major issue on the dashboard and Matt Self is up to date on how I'm tracking these ✌️ 

<img width="748" alt="Screenshot 2023-05-26 at 9 52 25 AM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/14154792/ded1a806-2e08-45e5-a261-f9dcae101763">

**Comment by saderagsdale** [comment]
@rocio-desantiago you're not working on this right? I'm moving it to the backlog.

**Comment by rocio-desantiago** [comment]
@saderagsdale my understanding there was a blocker on this but we already merged in a temporary fix in the meantime a couple of sprints ago

**Comment by joshkimux** [comment]
[If we were able to merge in the fix Jami explored](https://github.com/department-of-veterans-affairs/va.gov-team/issues/56080#issuecomment-1513204928), we can:
1. update the original ticket noting that change
2. close this ticket
3. I can create a bug ticket (or find an existing bug ticket) regarding the DST level issue to add this as a component that will need remediation in the future

**Comment by rocio-desantiago** [comment]
@joshkimux does the fix exist for checkboxes now? I think that was the previous blocker. But if the props are available for the checkbox, we can create a new ticket for this.

**Comment by joshkimux** [comment]
@rocio-desantiago let me ask Jami! I'll return back to this asap

**Comment by joshkimux** [comment]
@rocio-desantiago , @jamigibbs was kind enough to write us a DST ticket for this! I'm going to mark this as blocked until that one is resolved ❤️

**Comment by rocio-desantiago** [comment]
@joshkimux remind me what DST is 😄

**Comment by joshkimux** [comment]
@rocio-desantiago Design system team! I'll be sure to avoid acronyms next time 🙇

**Comment by saderagsdale** [comment]
@joshkimux Just to clarify, this is being handed to the design system team?

**Comment by joshkimux** [comment]
@saderagsdale yes— Jami opened up a ticket to create a new prop we need to complete this ticket. We will keep this open and blocked until that new prop is made by the design system team.

**Comment by HeatherWidmont** [comment]
Robin asking in platform channel when is the next version of the component library going to be available, that is currently blocking this ticket

**Comment by shiragoodman** [comment]
I read through the ticket and the comments, but can't easily identify which Design System component we're talking about here. @HeatherWidmont do you know?

**Comment by jamigibbs** [comment]
> Robin asking in platform channel when is the next version of the component library going to be available, that is currently blocking this ticket

@HeatherWidmont If you're talking about the [PR](https://github.com/department-of-veterans-affairs/component-library/pull/817) linked in this issue above that's related to the component library, that will be in our release going out today.

**Comment by Mottie** [comment]
I checked these changes in staging, and the `description` id is not being added to the input's `aria-describedby`. It _should_ be adding it, but it's not. This may take a bit more investigation.

**Comment by HeatherWidmont** [comment]
@jaredcunha do you have capacity to review this?

**Comment by jaredcunha** [comment]
I can confirm that this works. While I was unable to access the **Private Medical Record Authorization** page, but Robin had pointed to the **Notice of Evidence Page** which uses the same pattern. 

I've attached a screen and audio recording. As you can see, as I tabbed to the checkbox, the `aria-describedby` content is announced after the checkbox name.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/1094951/f266157d-48a9-4709-9247-4b54d7fd9a44

**Comment by Mottie** [comment]
@saderagsdale we can close this ticket now
